Symfony/CS/Fixer/Symfony/SpacesAfterSemicolonFixer.php

```diff
@@ -27,15 +27,22 @@ public function fix(\SplFileInfo $file, $content)
     {
         $tokens = Tokens::fromCode($content);
 
-        for ($index = count($tokens) - 3; $index > 0; --$index) {
-            if (!$tokens[$index]->equals(';') || $tokens[$index + 2]->isComment()) {
+        for ($index = 1, $count = count($tokens); $index < $count - 2; ++$index) {
+
+            // handle the exception case for the last `;` inside a `for` loop
+            if ($tokens[$index]->isGivenKind(T_FOR)) {
+                $index = $this->fixSemicolonsInFor($tokens, $index);
```

Comment:
is there any other case than a for-loop where the sequence `;)` could exist?
I don't think so. I think you do not need that special handling for `for` loop.
Just look for next meaningful token. if it is `)`, do not add a space.